### PR TITLE
Dependency fix

### PR DESCRIPTION
**Pull request by ombhilare999**
Hi,

I’ve integrated the `spdlogger` library into UGRAMM by including only its header files. After adding the `spdlogger` files to the UGRAMM project, I noticed that the binary size change is minimal.

Additionally, the Makefile has been further generalized and tested on the ECF server, where it works seamlessly out of the box.

For the Python scripts, I’ve added a `requirements.txt` file under the `scripts/` directory. These dependencies should be installed first in order to run the Python scripts on any platform.

Since this PR won’t interfere with other branches, I believe we can proceed with merging it into the master branch once everything checks out.

Regards,
Omkar Bhilare

### CODE REVIEW

**Comment by ombhilare999** [comment]
Hi @HamasWaqar ,
Can you review this please? Thank you.

**Comment by HamasWaqar** [comment]
F

> Hi,
> 
> I’ve integrated the `spdlogger` library into UGRAMM by including only its header files. After adding the `spdlogger` files to the UGRAMM project, I noticed that the binary size change is minimal.
> 
> Additionally, the Makefile has been further generalized and tested on the ECF server, where it works seamlessly out of the box.
> 
> For the Python scripts, I’ve added a `requirements.txt` file under the `scripts/` directory. These dependencies should be installed first in order to run the Python scripts on any platform.
> 
> Since this PR won’t interfere with other branches, I believe we can proceed with merging it into the master branch once everything checks out.
> 
> Regards, Omkar Bhilare

**Comment by HamasWaqar** [comment]
Other than that, everything else looks good. Let me know when you merge this branch, so I can submit my pull request for the locking.

**Comment by ombhilare999** [comment]
Hi @HamasWaqar ,

I have merged this PR, let me know once your changes related to `locking` are done.

File: scripts/requirements.txt [hunk 1]
```diff
@@ -0,0 +1 @@
+networkx==3.3
```
[the remark this answers is not in the document]
Comment (ombhilare999) [reply scripts/requirements.txt:1]: The spdlogger and json libraries that we are utilizing are header-only libraries, making them extremely easy to integrate with the UGRAMM project without increasing either the project size or the binary size of UGRAMM.

For the Python packages, it’s much more efficient to install them via pip, allowing users to obtain the latest version easily. We have provided a requirements.txt file, so they only need to execute the following command:
`pip install -r /path/to/requirements.txt`

The networkx library appears to be a complex dependency to include with the UGRAMM project; however, users will still need to install it via pip, which requires them to specify a local path in the pip install command. This process will also necessitate root access.

**Pull request merged**